pkg/utils/mainhelper/init.go

```diff
@@ -63,6 +67,9 @@ func Init(component string, opts ...InitOption) (context.Context, *InitRes) {
 	logger := logging.FromContext(ctx)
 	RunProfilingServer(ctx, logger, profilingHandler)
 
+	// Try starting the GCP Profiler if config is present.
+	ProcessGCPProfilerEnvConfigOrDie(component, logger)
```

Comment:
@liu-cong can you help adding a test for this? I'm not sure how the tests in init_test.go work.

Reply:
It seems that there is no way to check if the profiler is started or not. So I don't know how to test this.